tests/app/notifications/test_validators.py

```diff
@@ -61,50 +65,55 @@ def enable_redis(notify_api):
 def count_key(limit_type, service_id):
     if limit_type == "sms":
         return f"sms-{service_id}-2016-01-01-count"
+    elif limit_type == "email":
+        return f"email-{service_id}-2016-01-01-count"
     else:
         return f"{service_id}-2016-01-01-count"
 
 
 def near_key(limit_type, service_id):
     if limit_type == "sms":
         return f"nearing-daily-limit-sms-{service_id}-2016-01-01-count"
+    elif limit_type == "email":
+        return f"nearing-daily-email-limit-email-{service_id}-2016-01-01-count"
     else:
         return f"nearing-{service_id}-2016-01-01-count"
 
 
 def over_key(limit_type, service_id):
     if limit_type == "sms":
         return f"over-daily-limit-sms-{service_id}-2016-01-01-count"
+    elif limit_type == "email":
+        return f"over-daily-email-limit-email-{service_id}-2016-01-01-count"
     else:
         return f"over-{service_id}-2016-01-01-count"
 
 
-class TestCheckDailyLimits:
+class TestCheckDailySMSEmailLimits:
     @pytest.mark.parametrize(
-        "limit_type, key_type",
-        [("all", "test"), ("all", "team"), ("all", "normal"), ("sms", "test"), ("sms", "team"), ("sms", "normal")],
+        "limit_type",
+        ["email", "sms"],
     )
     def test_check_service_message_limit_in_cache_with_unrestricted_service_is_allowed(
-        self, notify_api, limit_type, key_type, sample_service, mocker
+        self, notify_api, limit_type, sample_service, mocker
     ):
         mocker.patch("app.notifications.validators.redis_store.get", return_value=1)
         mocker.patch("app.notifications.validators.redis_store.set")
         mocker.patch("app.notifications.validators.services_dao")
-
         if limit_type == "sms":
             with set_config(notify_api, "FF_SPIKE_SMS_DAILY_LIMIT", True):
                 check_sms_daily_limit(sample_service)
         else:
-            check_service_over_daily_message_limit(key_type, sample_service)
+            check_email_daily_limit(sample_service)
         app.notifications.validators.redis_store.set.assert_not_called()
         assert not app.notifications.validators.services_dao.mock_calls
 
     @pytest.mark.parametrize(
-        "limit_type, key_type",
-        [("all", "test"), ("all", "team"), ("all", "normal"), ("sms", "test"), ("sms", "team"), ("sms", "normal")],
+        "limit_type",
+        ["email", "sms"],
     )
     def test_check_service_message_limit_in_cache_under_message_limit_passes(
```

Comment:
More of a suggestion for later consideration:
Test names could be a little more specific, we don't really have a naming convention in Notify for our tests but perhaps something like: `given_service_under_message_limit_and_cache_key_exists_when_checking_daily_limit_then_key_fetched_from_redis`

It's long as hell but when I read this I immediately have a good grasp of what is under test.

Reply:
this is fair, we should standardize this going forward.